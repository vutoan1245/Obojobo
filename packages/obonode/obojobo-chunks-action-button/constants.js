const ACTION_BUTTON_NODE = 'ObojoboDraft.Chunks.ActionButton'

export { ACTION_BUTTON_NODE }
