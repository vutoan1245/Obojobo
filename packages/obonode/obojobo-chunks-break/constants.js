const BREAK_NODE = 'ObojoboDraft.Chunks.Break'
const NORMAL = 'normal'
const LARGE = 'large'

export { BREAK_NODE, NORMAL, LARGE }
