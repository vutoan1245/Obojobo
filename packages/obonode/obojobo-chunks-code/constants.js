const CODE_NODE = 'ObojoboDraft.Chunks.Code'
const CODE_LINE_NODE = 'ObojoboDraft.Chunks.Code.CodeLine'

export { CODE_NODE, CODE_LINE_NODE }
