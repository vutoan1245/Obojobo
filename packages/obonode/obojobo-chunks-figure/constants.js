const FIGURE_NODE = 'ObojoboDraft.Chunks.Figure'

export { FIGURE_NODE }
