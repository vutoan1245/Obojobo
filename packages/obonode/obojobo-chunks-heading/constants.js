const HEADING_NODE = 'ObojoboDraft.Chunks.Heading'

export { HEADING_NODE }
