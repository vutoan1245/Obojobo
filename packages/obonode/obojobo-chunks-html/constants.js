const HTML_NODE = 'ObojoboDraft.Chunks.HTML'

export { HTML_NODE }
