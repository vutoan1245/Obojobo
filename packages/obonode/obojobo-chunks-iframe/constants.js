const IFRAME_NODE = 'ObojoboDraft.Chunks.IFrame'

export { IFRAME_NODE }
