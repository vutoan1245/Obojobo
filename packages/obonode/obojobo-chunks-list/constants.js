const LIST_NODE = 'ObojoboDraft.Chunks.List'
const LIST_LINE_NODE = 'ObojoboDraft.Chunks.List.Line'
const LIST_LEVEL_NODE = 'ObojoboDraft.Chunks.List.Level'

const ALPHA = 'alpha'
const CIRCLE = 'circle'
const DECIMAL = 'decimal'
const DISC = 'disc'
const ORDERED = 'ordered'
const UNORDERED = 'unordered'
const SQUARE = 'square'
const LOWER_ALPHA = 'lower-alpha'
const LOWER_ROMAN = 'lower-roman'
const UPPER_ALPHA = 'upper-alpha'
const UPPER_ROMAN = 'upper-roman'

const unorderedBullets = [DISC, CIRCLE, SQUARE]
const orderedBullets = [DECIMAL, UPPER_ALPHA, UPPER_ROMAN, LOWER_ALPHA, LOWER_ROMAN]

export {
	LIST_NODE,
	LIST_LINE_NODE,
	LIST_LEVEL_NODE,
	ALPHA,
	CIRCLE,
	DECIMAL,
	DISC,
	ORDERED,
	UNORDERED,
	SQUARE,
	LOWER_ALPHA,
	LOWER_ROMAN,
	UPPER_ALPHA,
	UPPER_ROMAN,
	unorderedBullets,
	orderedBullets
}
