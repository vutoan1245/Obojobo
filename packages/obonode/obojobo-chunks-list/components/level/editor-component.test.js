import React from 'react'
import renderer from 'react-test-renderer'

import Level from './editor-component'
import { ALPHA, ORDERED, UNORDERED, SQUARE } from '../../constants'

describe('List editor', () => {
	test('Level component', () => {
		const component = renderer.create(
			<Level
				node={{
					data: {
						get: () => {
							return { bulletStyle: SQUARE, type: UNORDERED }
						}
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('Level component ordered', () => {
		const component = renderer.create(
			<Level
				node={{
					data: {
						get: () => {
							return { bulletStyle: ALPHA, type: ORDERED }
						}
					}
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
