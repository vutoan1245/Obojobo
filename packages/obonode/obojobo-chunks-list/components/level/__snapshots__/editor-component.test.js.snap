// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List editor Level component 1`] = `
<ul
  style={
    Object {
      "listStyleType": "square",
    }
  }
/>
`;

exports[`List editor Level component ordered 1`] = `
<ol
  style={
    Object {
      "listStyleType": "alpha",
    }
  }
/>
`;
